### DISCUSSION TITLE
我想知道为什么milvusClient的insert返回的result里面的id在我的环境上是每32个都是一样的 第33个才能有不同 是我的代码有问题吗

### DISCUSSION THREAD

**Discussion by ENg-122** [milvus-io/milvus#36762] [Q&A and General discussion] [answered]
const result = await client.insert({
        collection_name: DatasetVectorTableName,
        data: [
          {
            vector,
            teamId: String(teamId),
            datasetId: String(datasetId),
            collectionId: String(collectionId),
            createTime: Date.now()
          }
        ]
      });

      console.log("result1234", result)

      const insertId = (() => {
        if ('int_id' in result.IDs) {
          return `${result.IDs.int_id.data?.[0]}`;
        }
        return `${result.IDs.str_id.data?.[0]}`;
      })();

      return {
        insertId: insertId
      };

**Comment by xiaofan-luan** [comment]
what milvus version are u using?

**Reply by ENg-122** [reply]
milvusdb/milvus:v2.4.3

**Comment by zhanshuyou** [comment, marked as answer]
@ENg-122 
Hi, I tested your code and it seems to be working fine. One possible reason for the issue is the loss of numerical precision. If you're displaying this on the front end, try converting it to a String type.

![image](https://github.com/user-attachments/assets/da09b5a6-fc9a-4a63-a98d-73f54e3418dd)
![image](https://github.com/user-attachments/assets/b3a6f80d-07b8-4d31-a4fc-cb451029e2f5)

**Reply by ENg-122** [reply]
感谢大佬 确实是这个问题 数据精度损失了 改成varchar已经解决
